nano/node/wallet.cpp

```diff
@@ -1512,6 +1534,67 @@ void nano::wallets::reload ()
 	}
 }
 
+void nano::wallets::do_work_regeneration ()
+{
+	std::unique_lock<std::mutex> regeneration_lock (difficulty_mutex);
+	while (!stopped)
+	{
+		if (!difficulty_reque.empty ())
+		{
+			auto const first (difficulty_reque.begin ());
+			auto now (std::chrono::steady_clock::now ());
+			auto queued (first->first);
+			auto block (first->second);
+			std::shared_ptr<nano::block> block_l (block);
+			bool confirmed (false);
+			difficulty_reque.erase (first);
```

Comment:
The `regeneration_lock` could be released here (just make sure to lock again before the scope leaves)